### PR TITLE
add custom easyconfig parameter in GCC easyblock to control use of gold linker

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/easybuilders/easybuild-easyconfigs/issues/6097

### CODE REVIEW

File: easybuild/easyblocks/g/gcc.py [hunk 1]
```diff
@@ -434,7 +442,7 @@ def build_step(self):
 
                         # ensure generic build when 'generic' is set to True or when --optarch=GENERIC is used
                         # non-generic build can be enforced with generic=False if --optarch=GENERIC is used
-                        if build_option('optarch') == OPTARCH_GENERIC and self.cfg['generic'] != False:
+                        if build_option('optarch') == OPTARCH_GENERIC and self.cfg['generic'] is not False:
```
Comment (akesandgren) [inline easybuild/easyblocks/g/gcc.py]: is not False....
why not just just "self.cfg['generic']"
or at least "is True"

Comment (akesandgren) [reply easybuild/easyblocks/g/gcc.py]: Ahh, "use False to enforce non-generic regardless".... didn't notice that one.

Comment (boegel) [reply easybuild/easyblocks/g/gcc.py]: Indeed, it's easy to overlook that one, `False` is a special value here

**Comment by akesandgren** [comment]
@vanzod want to take a look too?

**Pull request merged**